### PR TITLE
add FQCN to M() references

### PR DESCRIPTION
**Pull request by bmillemathias**
##### SUMMARY
update also seealso links
https://github.com/ansible-collections/overview/issues/89

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
copy
shell
command

### CODE REVIEW

**Comment by samdoran** [comment]
I'm not sure we want to add `ansible.builtin` to all the modules in `ansible-base`. Is there a bigger reason behind this change?

**Comment by abadger** [comment]
> I'm not sure we want to add `ansible.builtin` to all the modules in `ansible-base`. Is there a bigger reason behind this change?

There is.... Here's a summary of a discussion that just occurred on IRC about this PR: https://github.com/ansible/community/issues/521#issuecomment-656399243  The discussions and decisions leading up to that are linked from that comment but you can definitely ping acozine, samccann, or me for more info.

**Comment by abadger** [comment]
I'm looking into the test failure.  I've found a few things out:

* The failure will no longer happen when https://github.com/ansible/ansible/pull/59761 is merged.  This is to be expected as the old syntax which doesn't understand the collection world is currently being generated.  59761 needs to be merged in order for the new syntax which understands collections to be used.
* The reason that only one plugin, lookup/fileglob.py, is generating an error is that it is the only non-module plugin being changed.
  * It looks like `make singlehtmldocs`, which the docs-build test uses, does not build html from the module docs (no wonder it's so much faster than make htmldocs!).  My guess is that all of our pages which are `:orphan:` are not built to html by `make singlehtmldocs`.
    * I'm not sure if I'll have a chance to fix this but I'll try converting the docs-build test to use `make htmldocs`on my collection-docs PR. Now that there's a ton less modules in core, that may be quick enough to run in CI.  If that looks like it will be too slow, I'll open a ticket instead.

File: lib/ansible/modules/expect.py [hunk 1]
```diff
@@ -66,9 +66,9 @@
...
     the response. The list functionality is new in 2.1.
-  - The M(expect) module is designed for simple scenarios. For more complex
-    needs, consider the use of expect code with the M(shell) or M(script)
-    modules. (An example is part of the M(shell) module documentation)
+  - The M(ansible.builtin.expect) module is designed for simple scenarios.
+    For more complex needs, consider the use of expect code with the M(ansible.builtin.shell)
+    or M(ansible.builtin.script) modules. (An example is part of the M(ansible.builtin.shell) module documentation)
 seealso:
 - module: script
 - module: shell
```
Comment (abadger) [inline lib/ansible/modules/expect.py]: These two need to be adjusted as well

**Comment by abadger** [comment]
Update on the test situation here: https://github.com/ansible/ansible/pull/59761#issuecomment-656418388

The additional files which could be fixed, similar to what you already have here are:
```
/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/add_host_module.rst:116: WARNING: undefined label:     ansible_collections.group_by_module (if the link has no caption the label must precede a section header)

/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/set_fact_module.rst:126: WARNING: undefined label:     ansible_collections.include_vars_module (if the link has no caption the label must precede a section header)

/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/tempfile_module.rst:142: WARNING: undefined label:     ansible_collections.file_module (if the link has no caption the label must precede a section header)
/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/tempfile_module.rst:144: WARNING: undefined label:     ansible_collections.win_tempfile_module (if the link has no caption the label must precede a section header)

/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/template_module.rst:544: WARNING: undefined label:     ansible_collections.copy_module (if the link has no caption the label must precede a section header)
/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/template_module.rst:546: WARNING: undefined label:     ansible_collections.win_copy_module (if the link has no caption the label must precede a section header)
/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/template_module.rst:548: WARNING: undefined label:     ansible_collections.win_template_module (if the link has no caption the label must precede a section header)

/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/wait_for_connection_module.rst:147: WARNING: undefined label: ansible_collections.wait_for_module (if the link has no caption the label must precede a section header)
/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/wait_for_connection_module.rst:149: WARNING: undefined label: ansible_collections.win_wait_for_module (if the link has no caption the label must precede a section header)
/var/tmp/vanilla/docs/docsite/rst/collections/ansible/builtin/wait_for_connection_module.rst:151: WARNING: undefined label: ansible_collections.win_wait_for_process_module (if the link has no caption the label must precede a section  header)
```

Additionally, there's two errors on parsing: 
* inventory script
* lookup url

But those are a different sort of error so I'll probably look into those myself.

File: lib/ansible/modules/expect.py (hunk 1 above)
Comment (bmillemathias) [reply lib/ansible/modules/expect.py]: fixed !

**Comment by bmillemathias** [comment]
ready_for_review

File: lib/ansible/modules/get_url.py [hunk 2]
```diff
@@ -17,15 +17,15 @@
        server I(must) have direct access to the remote resource.
      - By default, if an environment variable C(<protocol>_proxy) is set on
        the target host, requests will be sent through that proxy. This
-       behaviour can be overridden by setting a variable for this task
+      behaviour can be overridden by setting a variable for this task
```
Comment (Andersson007) [inline lib/ansible/modules/get_url.py]:
```suggestion
       behaviour can be overridden by setting a variable for this task
```

File: lib/ansible/modules/known_hosts.py [hunk 3]
```diff
@@ -14,8 +14,8 @@
 description:
    - The C(known_hosts) module lets you add or remove a host keys from the C(known_hosts) file.
    - Starting at Ansible 2.2, multiple entries per host are allowed, but only one for each key type supported by ssh.
-     This is useful if you're going to want to use the M(git) module over ssh, for example.
-   - If you have a very large number of host keys to manage, you will find the M(template) module more useful.
+     This is useful if you're going to want to use the M(ansible.builtin .git) module over ssh, for example.
```
Comment (Andersson007) [inline lib/ansible/modules/known_hosts.py]:
```suggestion
     This is useful if you're going to want to use the M(ansible.builtin.git) module over ssh, for example.
```

File: lib/ansible/modules/shell.py [hunk 4]
```diff
@@ -73,10 +73,10 @@
     version_added: "2.8"
 notes:
   - If you want to execute a command securely and predictably, it may be
-    better to use the M(command) module instead. Best practices when writing
-    playbooks will follow the trend of using M(command) unless the C(shell)
-    module is explicitly required. When running ad-hoc commands, use your best
-    judgement.
+    better to use the M(ansible.builtin.command) module instead. Best practices
+    when writing playbooks will follow the trend of using M(ansible.builtin.command)
+    unless the C(shell) module is explicitly required. When running ad-hoc
```
Comment (Andersson007) [inline lib/ansible/modules/shell.py]:
```suggestion
    unless the M(ansible.builtin.shell) module is explicitly required. When running ad-hoc
```

Comment (bmillemathias) [reply lib/ansible/modules/shell.py]: I was not sure about this one, I thought it was a configuration option as the `C()` implies

**Comment by Andersson007** [comment]
What about examles?
```
189 EXAMPLES = r'''
190 - name: Check that you can connect (GET) to a page and it returns a status 200
191   uri:
192     url: http://www.example.com
```
Should they use FQCNs as well? If yes, maybe it makes sense to change them here?

**Comment by Andersson007** [comment]
I grepped throug your branch to recheck. Found nothing to change except "EXAMPLES" and
```
docs/docsite/rst/user_guide/playbooks_vault.rst:      and some actions that deal with files (i.e M(copy), M(assemble), M(script), etc).
```

**Comment by bmillemathias** [comment]
> 
> 
> What about examles?
> ..
> Should they use FQCNs as well? If yes, maybe it makes sense to change them here?

Right I asked myself and I though to use this PR to fix link and I can do another PRs for doc examples

File: lib/ansible/modules/shell.py (hunk 4 above)
Comment (Andersson007) [reply lib/ansible/modules/shell.py]: unless the C(shell) `module` is explicitly required. I think the module is implied from the context

Comment (Andersson007) [reply lib/ansible/modules/shell.py]: not only because of the word `module`, from written a bit above

**Comment by Andersson007** [comment]
> > What about examles?
> > ..
> > Should they use FQCNs as well? If yes, maybe it makes sense to change them here?
> 
> Right I asked myself and I though to use this PR to fix link and I can do another PRs for doc examples

also a solution:) thanks

**Comment by abadger** [comment]
> > What about examles?
> > ..
> > Should they use FQCNs as well? If yes, maybe it makes sense to change them here?
> 
> Right I asked myself and I though to use this PR to fix link and I can do another PRs for doc examples

This sounds like a good plan.  And we'll want to get confirmation from @acozine and @samccann as to whether we want examples and `C()` use fqcn for ansible.builtin or not.

**Comment by Andersson007** [comment]
> > > What about examles?
> > > ..
> > > Should they use FQCNs as well? If yes, maybe it makes sense to change them here?
> > 
> > 
> > Right I asked myself and I though to use this PR to fix link and I can do another PRs for doc examples
> 
> This sounds like a good plan. And we'll want to get confirmation from @acozine and @samccann as to whether we want examples and `C()` use fqcn for ansible.builtin or not.

after we decide whether we want or not, we should also remember to change examples of EXAMPLES in dev_guide .rst files, e.g. in https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#examples-block
at least to mark that users must use proper FQCNs for collections (and ansible.builtin if we want).

**Comment by samccann** [comment]
Looks like things are passing now.  Any reason not to merge this now?

**Comment by bmillemathias** [comment]
> /rebuild_failed

don't loose time on the failing centos 6 build problem, this is a know issue.

**Comment by Andersson007** [comment]
there's not only centos6. osx too

**Review by Andersson007 — APPROVED** [review]
I think it's also unrelated
LGTM

**Review by felixfontein — APPROVED** [review]

**Review by samccann — APPROVED** [review]

**Pull request merged**